src/mender-update/http_resumer/http_resumer.cpp

```diff
@@ -309,6 +309,11 @@ void DownloadResumerAsyncReader::Cancel() {
 
 error::Error DownloadResumerAsyncReader::AsyncRead(
 	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end, io::AsyncIoHandler handler) {
+	if (eof_) {
```

Comment:
I think I might have come across this in one of my integration tests actually! Thx!